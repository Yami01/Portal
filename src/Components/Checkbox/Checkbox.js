/**
 *
 * Checkbox
 * @format
 * @flow
 *
 */

import type { AbstractComponent, Node } from "react"
import React, { memo } from "react"
import { Checkbox as RNCheckbox } from "react-native-paper"
import type { PropsType } from "./types"
import { Colors } from '@/Theme/Variables'

const Checkbox: AbstractComponent<PropsType> = memo(
  (props: PropsType): Node => {
    const { checked, onPress, color, uncheckedColor, disabled, testID } = props

    return (
      <RNCheckbox.Android
        status={checked ? 'checked' : 'unchecked'}
        onPress={onPress}
        color={color || Colors.dark}
        uncheckedColor={uncheckedColor || Colors.dark}
        disabled={disabled}
        testID={testID}
      />
    )
  },
)

export default Checkbox
