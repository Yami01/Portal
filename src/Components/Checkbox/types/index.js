/**
 *
 * @format
 * @flow
 *
 */

export type PropsType = {
  checked?: boolean,
  onPress: boolean => void,
  color?: string,
  uncheckedColor?: string,
  disabled?: boolean,
  testID?: string,
}
