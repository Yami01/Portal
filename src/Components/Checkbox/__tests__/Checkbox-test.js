/**
 *
 * Checkbox Test
 * @format
 * @flow
 *
 */

import React from 'react'
import renderer from 'react-test-renderer'
import { fireEvent, render } from '@testing-library/react-native'

import Checkbox from '../Checkbox'

describe('Checkbox', () => {
  const checkBoxTestId = 'checkBoxTestId'
  const onPressMock = jest.fn()

  test('Checkbox Render', () => {
    const tree = renderer.create(<Checkbox onPress={onPressMock} />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('Checkbox Press', () => {
    const { queryByTestId } = render(
      <Checkbox testID={checkBoxTestId} onPress={onPressMock} />,
    )

    const button = queryByTestId(checkBoxTestId)
    fireEvent.press(button)
    expect(onPressMock).toHaveBeenCalled()
  })
})
