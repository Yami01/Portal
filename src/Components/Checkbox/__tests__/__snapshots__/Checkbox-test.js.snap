// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox Checkbox Render 1`] = `
<View
  accessibilityLiveRegion="polite"
  accessibilityRole="checkbox"
  accessibilityState={
    Object {
      "checked": false,
      "disabled": false,
    }
  }
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "overflow": "hidden",
      },
      Object {
        "borderRadius": 18,
        "height": 36,
        "padding": 6,
        "width": 36,
      },
    ]
  }
>
  <View
    collapsable={false}
    style={
      Object {
        "transform": Array [
          Object {
            "scale": 1,
          },
        ],
      }
    }
  >
    <Text
      accessibilityElementsHidden={true}
      allowFontScaling={false}
      importantForAccessibility="no-hide-descendants"
      pointerEvents="none"
      selectable={false}
      style={
        Array [
          Object {
            "color": "#444",
            "fontSize": 24,
          },
          Array [
            Object {
              "lineHeight": 24,
              "transform": Array [
                Object {
                  "scaleX": 1,
                },
              ],
            },
            Object {
              "backgroundColor": "transparent",
            },
          ],
          Object {
            "fontFamily": "Material Design Icons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
    >
      ó°„±
    </Text>
    <View
      style={
        Array [
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          Object {
            "alignItems": "center",
            "justifyContent": "center",
          },
        ]
      }
    >
      <View
        collapsable={false}
        style={
          Object {
            "borderColor": "#444",
            "borderWidth": 0,
            "height": 14,
            "width": 14,
          }
        }
      />
    </View>
  </View>
</View>
`;
